foundation/org.eclipse.persistence.oracle/src/main/java/org/eclipse/persistence/platform/database/oracle/Oracle21Platform.java

```diff
@@ -0,0 +1,198 @@
+        // Mapping for JSON type is set in JsonTypeConverter#initialize.
+        fieldTypeMapping.put(JsonObject.class, new FieldTypeDefinition("JSON"));
+        fieldTypeMapping.put(JsonArray.class, new FieldTypeDefinition("JSON"));
+        fieldTypeMapping.put(JsonValue.class, new FieldTypeDefinition("JSON"));
+        return fieldTypeMapping;
+    }
+
+    /**
+     * INTERNAL
+     * Set the parameter in the JDBC statement at the given index.
+     * This support a wide range of different parameter types, and is heavily optimized for common types.
+     * Handles Postgres specific PGobject instances.
+     *
+     * @param parameter the parameter to set
+     * @param statement target {@code PreparedStatement} instance
+     * @param index index of the parameter in the statement
+     * @param session current database session
+     */
+    @Override
+    public void setParameterValueInDatabaseCall(
+            final Object parameter, final PreparedStatement statement,
+            final int index, final AbstractSession session
+    ) throws SQLException {
+        if (parameter instanceof OracleJsonValue) {
+            statement.setObject(index, parameter, OracleType.JSON);
+        } else {
+            super.setParameterValueInDatabaseCall(parameter, statement, index, session);
+        }
+    }
+
+    /**
+     * INTERNAL
+     * Set the parameter in the JDBC statement at the given index.
+     * This support a wide range of different parameter types, and is heavily optimized for common types.
+     * Handles Postgres specific PGobject instances.
+     *
+     * @param parameter the parameter to set
+     * @param statement target {@code CallableStatement} instance
+     * @param name name of the parameter in the statement
+     * @param session current database session
+     */
+    @Override
+    public void setParameterValueInDatabaseCall(
+            final Object parameter, final CallableStatement statement,
+            final String name, final AbstractSession session
+    ) throws SQLException {
+        if (parameter instanceof OracleJsonValue) {
+            statement.setObject(name, parameter, OracleType.JSON);
+        } else {
+            super.setParameterValueInDatabaseCall(parameter, statement, name, session);
+        }
+    }
+
+    // Oracle 21c specific JSON types support:
+    // Stores JsonValue instances as JSON.
+    /**
+     * INTERNAL:
+     * Convert JSON value field to JDBC statement type.
+     * Postgres JSON storage type is {@code JSONB} and target Java type is {@code PGobject}.
+     *
+     * @param <T> classification type
+     * @param jsonValue source JSON value field
+     * @return converted JDBC statement type
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T convertJsonValueToDataValue(final JsonValue jsonValue) throws PersistenceException {
+        if (jsonValue == null) {
+            return null;
+        }
+        final StringWriter sw = new StringWriter(128);
```

Comment:
It would be great to have create methods with jakarta.json.JsonValue/Object/Array on OracleJsonFactory.